### DISCUSSION TITLE
Add support for a GPU based simulator

### DISCUSSION THREAD

**Discussion by bilalshaikh42** [biosimulations/biosimulations#3185] [Ideas]
Found this paper about a gpu based simulator. 

https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009410#sec001

**Comment by jonrkarr** [comment]
I'm not familiar with the intersection of Docker and GPUs. Is there any special that needs to be considered? Also, the UConn HPC would need to have GPUs, as would any online platform (GitHub actions) we'd want to use to test tools.

**Comment by bilalshaikh42** [comment]
Yes, this would be an involved change. Github actions is adding support for
gpus in the near future.
The hpc probably has some gpus, but this would need to be requested during
job submission.

**Comment by jonrkarr** [comment]
It is now possible to submit HPC jobs with specific hardware requirements. If we captured requirements of simulation tools, this could be linked together via Slurm constraints. In the middle, we need a standardized vocabulary for describing constraints. UConn HPC uses [this](https://github.com/CBC-UCONN/Slurm_Feature_Constraints). We could potentially turn that into an enumeration or something we treat like an ontology.
